### DISCUSSION TITLE
Error trace beetwen rear ports

### DISCUSSION THREAD

**Discussion by romanmendelproject** [netbox-community/netbox#7275] [General]
I run a netbox using docker, when building a trace between two devices via rear ports, I get an error: 'NoneType' object has no attribute 'origin'
![trace_error](https://user-images.githubusercontent.com/33754158/133561145-065ba1d6-c373-497b-8196-2d72a7af663a.png)

**Comment by candlerb** [comment]
Cable traces via frontports/rearports works for me, so can you give the exact steps to reproduce?  What devices/frontports/rearports and what cables did you put between them?

Also, can you provide a backtrace for the exception (set DEBUG=True in configuration.py)

**Comment by romanmendelproject** [comment]
At first I thought that the problem might be in my system after migrating to a new version, but then I brought up the new system on a different server and the error repeated itself.

I have filled in only the required attributes.
![1](https://user-images.githubusercontent.com/33754158/133563606-2ac97f9c-83b6-4ea5-be6d-0aa538bb0860.png)
![2](https://user-images.githubusercontent.com/33754158/133563629-23299a38-b706-40ba-a883-63390a06f089.png)
![3](https://user-images.githubusercontent.com/33754158/133563659-f16d7d8a-e8eb-4bdc-8145-8009be59a23e.png)
![5](https://user-images.githubusercontent.com/33754158/133563673-ecd0258e-9be1-403d-ae61-527c3289d9ca.png)


Environment:


Request Method: GET
Request URL: http://192.168.0.1/dcim/rear-ports/8597/trace/

Django Version: 3.2.7
Python Version: 3.9.5
Installed Applications:
['django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.humanize',
 'corsheaders',
 'debug_toolbar',
 'graphiql_debug_toolbar',
 'django_filters',
 'django_tables2',
 'django_prometheus',
 'graphene_django',
 'mptt',
 'rest_framework',
 'taggit',
 'timezone_field',
 'circuits',
 'dcim',
 'ipam',
 'extras',
 'tenancy',
 'users',
 'utilities',
 'virtualization',
 'django_rq',
 'drf_yasg',
 'discover_interfaces_plugin.DiscoverInterfacesConfig']
Installed Middleware:
['graphiql_debug_toolbar.middleware.DebugToolbarMiddleware',
 'django_prometheus.middleware.PrometheusBeforeMiddleware',
 'corsheaders.middleware.CorsMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'netbox.middleware.ExceptionHandlingMiddleware',
 'netbox.middleware.RemoteUserMiddleware',
 'netbox.middleware.LoginRequiredMiddleware',
 'netbox.middleware.APIVersionMiddleware',
 'netbox.middleware.ObjectChangeMiddleware',
 'django_prometheus.middleware.PrometheusAfterMiddleware']



Traceback (most recent call last):
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/views/generic/base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)
  File "/opt/netbox/netbox/dcim/views.py", line 2392, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/opt/netbox/netbox/utilities/views.py", line 93, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/opt/netbox/venv/lib/python3.9/site-packages/django/views/generic/base.py", line 98, in dispatch
    return handler(request, *args, **kwargs)
  File "/opt/netbox/netbox/netbox/views/generic.py", line 71, in get
    **self.get_extra_context(request, instance),
  File "/opt/netbox/netbox/dcim/views.py", line 2418, in get_extra_context
    api_viewname = f"{path.origin._meta.app_label}-api:{path.origin._meta.model_name}-trace"

Exception Type: AttributeError at /dcim/rear-ports/8597/trace/
Exception Value: 'NoneType' object has no attribute 'origin'

**Reply by candlerb** [reply]
Ah, so you're:
(a) starting the trace at a rearport;
(b) the corresponding frontport is not connected

I now see the same if I start a trace from the rearport of a patchpanel where the frontport isn't in use.

That's a nice reproducer, so I suggest you open this as a bug report.

**Reply by romanmendelproject** [reply]
![6](https://user-images.githubusercontent.com/33754158/133570340-66bb3b83-901f-4140-8532-c412399743bb.png)
> Ah, so you're:
> (a) starting the trace at a rearport;
> (b) the corresponding frontport is not connected
> 
> I now see the same if I start a trace from the rearport of a patchpanel where the frontport isn't in use.
> 
> That's a nice reproducer, so I suggest you open this as a bug report.

Thanks. In the old version 2, you could start tracing from anywhere.
Also, if my interface is connected to the front port of the patch panel, the rear port is connected to the rear port of another patch panel, and the second patch panel is not connected by the front port, then it builds tracing, but does not render.
This way I can get a normal trace if I have interfaces on both sides
